kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/NonJVMClientsST.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests;
+
+import java.time.Duration;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.strimzi.api.kafka.model.kafka.Kafka;
+
+import io.kroxylicious.systemtests.extensions.KroxyliciousExtension;
+import io.kroxylicious.systemtests.installation.kroxylicious.Kroxylicious;
+import io.kroxylicious.systemtests.steps.KafkaSteps;
+import io.kroxylicious.systemtests.steps.KroxyliciousSteps;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaNodePoolTemplates;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaTemplates;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * The Kroxylicious system tests.
+ */
+@ExtendWith(KroxyliciousExtension.class)
+class NonJVMClientsST extends AbstractST {
```

Comment:
I understand your motivations here and this could be a good first step towards our end goal but I'm worried its driving things in the wrong direction.

Ultimately our goal for the proxy is to be client agnostic so I would sort of hope that we could run every system test with every Kafka client we support (for a loose definition of support) I see the client as one dimension in a test matrix. We also want to ensure messages produced with one client are consumed by consumers of a different client (this should be something we can take for granted but its our implicit promise in being a transparent proxy). However if we go down the path of exposing `KroxyliciousSteps.produceMessagesWithKcat` then we have encoded the client support in the API which makes varying it hard.

Reply:
I'm thinking of having a config file with test clients supported and running all tests for each client if needed. 
* java client
* kcat
* kaf

By default we can set only Java clients for each PR (to save time) and all clients for each merge and release pipelines. But this could be a bigger feature that could be in another PR that will affect as well to the pipelines configuration. WDYT?

Reply:
I think we would still want to smoke test every PR with the every client.

Reply:
> I think we would still want to smoke test every PR with the every client.

+0.5.  There are always practical issues that mean some tests need to be excluded for some clients as not all CLIs have the same feature set.  The fact we are gesturing through a command line always puts limits on the possible.

I think the change @robobario is suggesting above, to put the test clients behind an interface would be a step in this direction.  I'd suggest making that one as part of this work, and come back to the test matrix idea in a separate step.

Reply:
Yeah to be clear I wasn't intentionally suggesting we should go further in this PR I was just trying to outline the longer term vision and trying to shape the API in that direction.

Reply:
I'm currently working on having the idea suggested by @robobario, that looks great to me. Then we can play more easily with test clients to be adapted to the tests producing and consuming what it is needed for each test.